packages/gatsby-theme-api-docs/gatsby-node.js

```diff
@@ -9,11 +9,62 @@ const fs = require('fs');
 // Ensure that certain directories exist.
 // https://www.gatsbyjs.org/tutorial/building-a-theme/#create-a-data-directory-using-the-onprebootstrap-lifecycle
 exports.onPreBootstrap = ({ reporter }) => {
-  const requiredDirectories = ['src/api-specs'];
+  const requiredDirectories = ['src/api-specs', 'src/data'];
   requiredDirectories.forEach((dir) => {
     if (!fs.existsSync(dir)) {
       reporter.info(`creating the ${dir} directory`);
       fs.mkdirSync(dir);
     }
   });
 };
+
+exports.createSchemaCustomization = ({ actions, schema }) => {
+  actions.createTypes(
+    schema.buildObjectType({
+      name: 'TypeLocations',
+      fields: {
+        type: { type: 'String!' },
+        path: { type: 'String!' },
+      },
+      interfaces: ['Node'],
+    })
+  );
+};
+
+exports.onCreateNode = ({
+  node,
+  getNode,
+  actions,
+  createNodeId,
+  createContentDigest,
+}) => {
+  const parent = getNode(node.parent);
+  const isTypeLocationsFile =
+    parent &&
+    parent.sourceInstanceName === 'type-locations' &&
+    parent.internal.mediaType === 'text/yaml';
+
+  if (!isTypeLocationsFile) {
+    return;
+  }
+
+  const typeLocationData = {
+    // The name of the file
+    path: node.path,
+  };
+
+  actions.createNode({
+    ...typeLocationData,
+    // Required fields
+    id: createNodeId(`${node.id} >>> Type location`),
+    parent: node.id,
+    children: [],
+    internal: {
+      type: `Type location`,
```

Comment:
Not convinced of the Type name with a space - it will appear in the GraphQL API that way or auto-slugified

UpperCamelCase is the current convention: https://github.com/commercetools/commercetools-docs-kit/blob/main/packages/gatsby-theme-docs/gatsby-node.js#L283